### DISCUSSION TITLE
How can I upload a CSV file with non-UTF8 character set

### DISCUSSION THREAD

**Discussion by ethanhuang1009** [apache/superset#18434] [Ideas]
### Expected results
upload success

### Actual results
upload failed and I faced an error without any solutions

#### Screenshots
<img width="661" alt="微信截图_20210203181452" src="https://user-images.githubusercontent.com/65382362/106733116-70be4f00-664c-11eb-93e8-7e2b1e844f0c.png">

**Unable to upload CSV file "11.csv" to table "11" in database "test". Error message: 'utf-8' codec can't decode byte 0xd4 in position 0: invalid continuation byte**

#### How to reproduce the bug

1. Go to 'upload a csv'
2. Click on 'select a file.'
4. See an error

### Environment

(please complete the following information):

- superset version: `1.0.0`
- python version: `python3.8`
- node.js version: `node -v`

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by zhaoyongjie** [comment]
@ethanhuang1009 
Please check the CSV file enconding.
1. Do not use CSV file exported from Excel on Windows plantform, Excel exported CSV file encoding is `utf8-with-bom`. 
2. The Superset only supports utf8 encoding CSV file import, so please carefully check the file encoding format.

This is a rough encoding issue,  in the future we can use some solution guess that file encoding.

======================
In Chinese
抱歉, 这是个 CSV编码问题.
1. 不要直接使用 Excel 导出的CSV, 他是 UTF8-bom 格式.
2. 当前只支持 UTF8 格式导入, 其他比如 CP936(GB2312) 等没有很好支持. 
这是个比较麻烦的编码问题, 您检查下文件编码格式.如果是 UTF8 一定可以. 你也可以使用 SublimeText 这类工具转换一下编码.

**Comment by ethanhuang1009** [comment]
@zhaoyongjie  Thanks you very much. I change the file encoding format from **utf8-with-bom** to utf8, then I can upload the file successfully. And I hope Superset can find and give us a better solution for the non-utf8 character set files in the near future.
